src/mesh.cpp

```diff
@@ -587,23 +590,19 @@ RegularMesh::RegularMesh(pugi::xml_node node)
     }
 
     // Set width
-    if (shape_.size() > 0) {
-      width_ = xt::eval((upper_right_ - lower_left_) / shape_);
-    }
+    width_ = xt::eval((upper_right_ - lower_left_) / shape_);
   } else {
     fatal_error("Must specify either <upper_right> and <width> on a mesh.");
   }
 
   // Make sure lower_left and dimension match
-  if (shape_.size() > 0) {
-    if (shape_.size() != lower_left_.size()) {
-      fatal_error("Number of entries on <lower_left> must be the same "
-        "as the number of entries on <dimension>.");
-    }
-
-    // Set volume fraction
-    volume_frac_ = 1.0/xt::prod(shape_)();
+  if (shape_.size() != lower_left_.size()) {
```

Comment:
I would move this up to where we read lower_left, because before this line, we check that `lower_left_` is the same length as `upper_right_`, but in theory both could be wrong and that check wouldn't see it (so you might have problems evaluating `width_` if the shapes of `upper_right_` and `lower_left_` don't match `shape_`.

Reply:
Good point -- I'll move that